src/feature.js

```diff
@@ -87,41 +85,34 @@ class FxABrowserFeature {
   _noUser() {
     this._defaultAvatar();
     browser.browserAction.setIcon({ path: "icons/avatar.svg" });
-    browser.browserAction.setPopup({
-      popup: "popup/sign_in/sign_in.html",
-    });
+    browser.browserAction.setPopup({ popup: "popup/sign_in/sign_in.html" });
   }
 
   _unverifiedUser() {
     this._defaultAvatar();
     browser.browserAction.setIcon({ path: "icons/avatar_confirm.svg" });
-    browser.browserAction.setPopup({
-      popup: "popup/unverified/unverified.html",
-    });
+    browser.browserAction.setPopup({ popup: "popup/unverified/unverified.html" });
   }
 
   _verifiedUser(user) {
-    if (!user.profileCache || !user.profileCache.profile.avatar) {
-      this._defaultAvatar(user.email);
+    if (!user || user.avatarDefault) {
+      this._defaultAvatar();
     } else {
-      this._userAvatar(user.profileCache.profile.avatar);
+      this._userAvatar(user.avatar);
     }
 
-    browser.browserAction.setPopup({
-      popup: "popup/menu/menu.html",
-    });
+    browser.browserAction.setPopup({ popup: "popup/menu/menu.html" });
   }
 
   _defaultAvatar() {
     if (this._avatar) {
       this._avatar = null;
-      // TODO: Render first character of email instead
-      browser.browserAction.setIcon({ path: "icons/avatar.svg" });
     }
+    // TODO We should also handle rendering first letter of email as the avatar in here.
+    browser.browserAction.setIcon({ path: "icons/avatar.svg" });
```

Comment:
@philbooth Yea, I added these extra checks because `this._avatar` was storing the image data, and would skip reloading the avatar if user uploaded a new one. Checking the against the url helped to prevent having a stale image. Might be a better solution, but I am drawing a blank at the moment.

Reply:
Got it, makes sense. I think we can still have similar logic in that case, maybe just wrap it up in a little method so it's obvious to readers/future us what's going on. Happy to PR something tomorrow if you don't beat me to it.

(so definitely merge this in the meantime, I mean)